android/src/com/mapswithme/maps/MwmActivity.java

```diff
@@ -383,8 +383,7 @@ public void showDownloader(boolean openDownloaded)
     args.putBoolean(DownloaderActivity.EXTRA_OPEN_DOWNLOADED, openDownloaded);
```

Comment:
@arnaudvergnet, thanks for contribution!
Is this patch a pure refactoring? Does it change UX? Please provide a video if UX is also changed. 
I've installed a beta version with your patch to one of my devices for testing. Let's see.

Reply:
By refactoring, I also fixed some small issues like toolbars appearing over one another, and Android back button not stopping search while in navigation. So it changed UX but only to fix bugs.

I still have more testing to do based on the list provided by @biodranik, I'll try to keep UX changes to a minimum and will provide a video of what changed when testing is done.

Reply:
Thanks a lot for explanation!